server/go.mod

```diff
@@ -46,6 +46,11 @@ require (
 	gorm.io/gorm v1.21.15
 )
 
+require (
+	golang.org/x/oauth2 v0.15.0 // indirect
+	mpldr.codes/oidc v0.0.0-20231223203712-a59dee5fc440
```

Comment:
The package proxy kicks in. Worst case scenario is that there has to be a replace directive in the go.mod that points to the backing repo over at https://git.sr.ht/~mpldr/oidc

The main reason for writing it was to make OIDC stupid easy to use, since I've run into it a few times now. OIDC is basically oauth2, but the token contains extra details by default and the plan is to expose them in an easy to use manner.
Basically: you shouldn't have to understand how OIDC is implemented, what the tokens are signed with, and all that fun stuff.